seed/models/data_quality.py

```diff
@@ -697,24 +716,28 @@ def _check(self, rules, row):
             label_applied = False
             display_name = rule.field
 
-            if hasattr(row, rule.field):
-                value = getattr(row, rule.field)
-                # TODO cleanup after the cleaner is better able to handle fields with units on import
-                # If the rule doesn't specify units only consider the value for the purposes of numerical comparison
-                if isinstance(value, ureg.Quantity) and rule.units == '':
-                    value = value.magnitude
-            else:  # rule is for extra_data
-                value = row.extra_data.get(rule.field, None)
-
-                if ' (Invalid Footprint)' in rule.field and value is not None:
-                    self.add_invalid_geometry_entry_provided(row.id, rule, display_name, value)
-                    continue
-
-                try:
-                    value = rule.str_to_data_type(value)
-                except DataQualityTypeCastError:
-                    self.add_result_type_error(row.id, rule, display_name, value)
-                    continue
+            if rule.for_derived_column:
+                derived_column = derived_columns_by_name[rule.field]
+                value = derived_column.evaluate(inventory_state=row)
```

Comment:
Good to see it was just a matter of adding the values of derived columns to each "record", and it doesn't seem necessary to update the code/logic that inspects the values.